Projects/Spaceman/spaceman.py

```diff
@@ -1,68 +1,132 @@
-    guessed correctly, the string should contain the letter at the correct position.  For letters
-    in the word that the user has not yet guessed, shown an _ (underscore) instead.
+    A function that is used to get a string showing the letters guessed so far in the secret word and underscores for letters that have not been guessed yet.
+
+    Args: 
+        secret_word (string): the random word the user is trying to guess.
+        letters_guessed (list of strings): list of letters that have been guessed so far.
+
+    Returns: 
+        string: letters and underscores.  For letters in the word that the user has guessed correctly, the string should contain the letter at the correct position.  For letters in the word that the user has not yet guessed, shown an _ (underscore) instead.
     '''
-    # FILL IN YOUR CODE HERE...
 
+    #TODO: Loop through the letters in secret word and build a string that shows the letters that have been guessed correctly so far that are saved in letters_guessed and underscores for the letters that have not been guessed yet
 
+    pass
 
 
 def get_available_letters(letters_guessed):
     '''
-    lettersGuessed: list of letters that have been guessed so far
-    returns: string, comprised of letters that represents what letters have not
-      yet been guessed.
+    A function that is used to get the letters that have not been guessed yet.
+
+    Args:
+        letters_guessed (list of strings): list of letters that have been guessed so far
+
+    Retunrs: 
+        string: letters that represent what letters have not yet been guessed.
+    '''
+
+    alphabet = "abcdefghijklmnopqrstuvwxyz"
+
+    #TODO: Loop through the letters of the alphabet and build a string of letters that are not in letters_guessed
+
+    pass
+
+
+def is_guess_in_word(guess, secret_word):
+    '''
+    A function to check if the guessed letter is in the secret word
+
+    Args:
+        guess (string): The letter the player guessed this round
+        secret_word (string): The secret word
+
+    Returns:
+        bool: True if the guess is in the secret_word, False otherwise
+
     '''
+    #TODO: check if the letter guess is in the secret word
+
+    pass
 
 
 
 
 def spaceman(secret_word):
     '''
     secretWord: string, the secret word to guess.
-
     Starts up a game of Spaceman in the command line.
-
     * At the start of the game, let the user know how many
       letters the secretWord contains.
-
     * Ask the user to guess one letter per round.
-
     * The user should receive feedback immediately after each guess
       about whether their guess appears in the computer's word.
-
     * After each round, you should also display to the user the
       partially guessed word so far, as well as letters that the
       user has not yet guessed.
     '''
-    # FILL IN YOUR CODE HERE...
 
 
-#
-# secret_word = load_word()
-# spaceman(load_word())
+    letters_guessed = []
```

Comment:
from here through line 124, is this possibly too much scaffolding? do we want to give them less to work with and figure out on their own? I can see arguments either way, I just wonder if we're now giving too much away. thoughts?

Reply:
I think it would be better to have less scaffolding, I will condense the todo's to be more general and remove the code I added under the main spaceman function